pavelib/assets.py

```diff
@@ -89,6 +90,36 @@ def applicable_sass_directories(systems=None):
     return applicable_directories
 
 
+class DebounceEventQueue(EventQueue):
+    """
+    Event queue that discards puts if they happen too quickly.
+    """
+    wait_seconds = 1
+
+    def __init__(self, *args, **kwargs):
+        EventQueue.__init__(self, *args, **kwargs)
+        self._last_updated = None
+
+    def _put(self, *args, **kwargs):
+        """
+        Only puts the item if the last put was at least `wait_seconds` ago.
+        """
+        now = datetime.now()
+        if not self._last_updated or (now - self._last_updated).total_seconds() > self.wait_seconds:
+            self._last_updated = now
+            EventQueue._put(self, *args, **kwargs)
+
+
+class DebouncePollingObserver(PollingObserver):
```

Comment:
It would be nice to not use polling mode if we know that another mode will work. Perhaps the choice of observer could be overridable, with polling as the default?